otel-extension/src/main/scala/io/scalac/mesmer/otelextension/instrumentations/zio/advice/ZIOCounterAdvice.scala

```diff
@@ -0,0 +1,12 @@
+package io.scalac.mesmer.otelextension.instrumentations.zio.advice
+
+import net.bytebuddy.asm.Advice
+import zio.metrics.Metric.Counter
+
+import io.scalac.mesmer.otelextension.instrumentations.zio.ZIOMetrics
+
+object ZIOCounterAdvice {
+  @Advice.OnMethodExit
+  def counter(@Advice.Argument(0) zioMetricName: String, @Advice.Return counter: Counter[_]): Unit =
+    ZIOMetrics.registerCounterAsyncMetric(zioMetricName, counter)
```

Comment:
Why do we use async metrics here? IMO it would be better to use sync ones and avoid issue with porting histograms. We should be able to do so overriding `zio.metrics.Metric.fromMetricKey` to update otel instrument as well as zio counter. I think this method should be also portable for other metrics types supported in zio